### PR TITLE
[SYCL] Add SYCL 2020 info::device::built_in_kernel_ids support

### PR DESCRIPTION
**Pull request by dnmokhov**
- Add `ProgramManager::getBuiltInKernelID`, which generates and caches built-in kernel IDs.
- Use that API to generate or look up built-in kernel IDs, when queried.
- Throw an exception in program manager when actually trying to use built-in kernels, since they are not yet fully supported.
- Add SYCL 2020 deprecation warning for `built_in_kernels` (old query).

Test: intel/llvm-test-suite#576

### CODE REVIEW

File: sycl/source/detail/device_info.hpp
Comment (vladimirlaz) [inline sycl/source/detail/device_info.hpp:289]: I think we will get the deprecation warning for info::device::built_in_kernels here.

Comment (dnmokhov) [reply sycl/source/detail/device_info.hpp:289]: Since this is not a user-visible include file but is in `sycl/source`, our compiler should never be compiling this code, right?

Comment (vladimirlaz) [reply sycl/source/detail/device_info.hpp:289]: It is better to avoid warnings in RT library builds as well.

**Review by vladimirlaz — APPROVED** [review]

**Comment by dm-vodopyanov** [comment]
Waiting for CI results from https://github.com/intel/llvm-test-suite/pull/576#issuecomment-975274535 before merging.

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**
